typescript-json-schema.ts

```diff
@@ -584,7 +584,11 @@ export class JsonSchemaGenerator {
             }
             if (this.args.generateRequired) {
                 const requiredProps = props.reduce((required: string[], prop: ts.Symbol) => {
-                    if (!(prop.flags & ts.SymbolFlags.Optional) && !(<any>prop).mayBeUndefined) {
+                    if (!(prop.flags & ts.SymbolFlags.Optional)
+                        && !(<any>prop).mayBeUndefined
+                        && prop["tags"].find((x: any): boolean => {
```

Comment:
Sure, how's that look?

Also, let me know when you're ready to merge and I'll squash my commits.